### ISSUE TITLE
XHR NetworkError with Jellyfin

### ISSUE THREAD

**Issue by Nick-NCSU** [MALSync/MALSync#2486]
### Describe the Bug

After updating to Manifest V3, the Jellyfin feature no longer seems to work on Firefox Desktop (it continues to work fine on Chrome).

Per the browser console I found the error to come from the following lines:
```sh
MAL-Sync Api Call http://192.168.1.230:8096/System/Info?api_key=[REDACTED]
uncaught exception: NetworkError when attempting to fetch resource. 
```
which I have tracked to come from the following line:
https://github.com/MALSync/MALSync/blob/2ace572e5b843cba852f80a89d064c06fda4815c/src/pages/Jellyfin/main.ts#L399

After some testing, I was able to resolve this issue by replacing the XHR call with fetch.
```diff
- return api.request.xhr('GET', url).then(res => {
+ return fetch(url).then(res => {
...
- JSON.parse(response.responseText);
+ await response.json();
```

### Steps to reproduce the behavior

Using Firefox Desktop play an Anime on Jellyfin.

### Browser

Firefox Desktop

### Type

Extension

### URL

Local jellyfin install + custom domain with port removed (http://192.168.1.230/*)

**Comment by lolamtisch** [comment]
With MV3 in firefox, requests from extensions start to respect CORS, if they do not have the permissions for that. That is what I suspect here. You can check it that is really is the case by opening the debugging page `about:debugging#/runtime/this-firefox` then click on inspect on malsync and then open the network tab. Reopening jellyfill should then show the request there. We added the option `Host Permissions` to be able to add these kind of permissions, but theoretically by adding the custom domain for jellyfin, it should theoretically already have the permission, not sure if it is a ip+port problem that breaks here.

**Comment by Nick-NCSU** [comment]
As you suspected I am seeing the call fail due to CORS.

As for permissions, MAL-Sync has both `http://192.168.1.230` and `<all_urls>` enabled.

**Comment by henrik9999** [comment]
Is the request going to http://192.168.1.230

**Comment by Nick-NCSU** [comment]
Ah, that explains it. Looks like the request is going to `https://192.168.1.230/`

**Comment by lolamtisch** [comment]
Try with `*://192.168.1.230/*)`  then

**Comment by lolamtisch** [comment]
Can you enter `ApiClient` in the console on jellyfin, not in malsync, and check what the `_serverAddress` contains? Because that is the url we get from jellyfin to use for the requests.

**Comment by Nick-NCSU** [comment]
It contains `http://192.168.1.230:8096/` which explains why switching from XHR to fetch successfully makes the calls. 

It looks like, for some reason, the XHR call is changing `http` to `https`.

**Comment by lolamtisch** [comment]
Can you see in the requests if there was a redirect or if it was a preflight request?

Not sure if that is related here, could you check the state of this option?
https://support.mozilla.org/en-US/kb/https-only-prefs

**Comment by Nick-NCSU** [comment]
I believe I have found the issue. It seems like the [default CSP](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Content_Security_Policy#default_content_security_policy) in MV3 in Firefox for extensions is `"script-src 'self'; upgrade-insecure-requests;"`. This is different than the default on [Chrome](https://developer.chrome.com/docs/extensions/reference/manifest/content-security-policy#default_policy).

The `upgrade-insecure-requests` ([docs](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Content_Security_Policy#default_content_security_policy)) seems to be what is causing this issue. It sounds like the only way to fix this issue would be to disable that policy.

I have confirmed that adding this to the manifest.json resolves the issue and have opened a PR here: #2488 
```json
"content_security_policy": {
    "extension_pages": "script-src 'self'; object-src 'self';"
  }
```
